argo/providers/kfp/metadata_store.go

```diff
@@ -58,21 +49,60 @@ func (gms *GrpcMetadataStore) GetServingModelArtifact(ctx context.Context, workf
 		return nil, err
 	}
 
-	results := make([]common.ServingModelArtifact, 0)
+	results := make([]common.Artifact, 0)
 	for _, artifact := range artifactsResponse.GetArtifacts() {
-		if artifact.GetTypeId() == artifactTypeId {
-			artifactUri := artifact.GetUri()
-			artifactName := artifact.GetCustomProperties()[ArtifactNameCustomProperty].GetStringValue()
-			modelHasBeenPushed := artifact.GetCustomProperties()[PushedCustomProperty].GetIntValue()
-
-			if artifactName != "" && artifactUri != "" && modelHasBeenPushed == 1 {
-				results = append(results, common.ServingModelArtifact{
-					Name:     artifactName,
-					Location: artifactUri,
-				})
+		artifactUri := artifact.GetUri()
+		artifactName := artifact.GetCustomProperties()[ArtifactNameCustomProperty].GetStringValue()
+		if artifactUri == "" {
+			continue
+		}
+
+		for _, artifactDef := range artifactDefs {
```

Comment:
move this loop outside to reduce the number of time evaluator needs to be created